rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -455,37 +459,70 @@ def _tag_id_index_mapping_for(
 
         return tag_id_dict
 
-    @staticmethod
-    def _find_example_for_label(
-        label: Text, examples: List[Message], attribute: Text
-    ) -> Optional[Message]:
-        for ex in examples:
-            if ex.get(attribute) == label:
-                return ex
-        return None
-
     def _check_labels_features_exist(
         self, labels_example: List[Message], attribute: Text
     ) -> bool:
-        """Checks if all labels have features set."""
+        """Checks whether there is at least one kind of feature per label.
 
+        Only checks whether one of the configured relevant featurizers has attached
+        some feature to each of the labels.
```

Comment:
except there might be features attached but we ignore those (because they're from featurizers we're not interested in) - can try to squeeze that in first line

Reply:
removed this method and test completely because it would just check whether label_example.features_present (and "all") works correctly and there is a test for that already